### PR TITLE
Enable Wire 2 for Internal SSH Server

### PR DESCRIPTION
**Pull request by Gusted**
- Git only decides to use the Wire 2 protocol when `git {receive,upload}-pack` receive the `GIT_PROTOCOL` environment with as value `version=2`. Currently the internal SSH Server wasn't passing this environment through. The `gitea serv` code already passed all received environments to the git command, so no code changes there.
- This is mentioned in Git manual, https://git-scm.com/docs/git#Documentation/git.txt-codeGITPROTOCOLcode

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Comment by Gusted** [comment]
Btw just for sanity check, this somehow was enabled for Git SmartHTTP. However that PR includes a regex check, should similar be done here? I cannot think of any attack surface of why you should sanitize this specific environment. https://github.com/go-gitea/gitea/pull/12170/files#diff-146b4048e1c4fd5339ff4d6d49af9f663a1add738588512eb25f01ca5702f2acR487

**Review by noerw — APPROVED** [review]

File: modules/ssh/ssh.go [hunk 1]
```diff
@@ -75,11 +75,21 @@ func sessionHandler(session ssh.Session) {
 	ctx, cancel := context.WithCancel(session.Context())
 	defer cancel()
 
+	gitProtocol := ""
+	for _, env := range session.Environ() {
+		if strings.HasPrefix(env, "GIT_PROTOCOL=") {
+			// The value would be version=2, so using normal split doesn't work here.
+			gitProtocol = strings.SplitN(env, "=", 2)[1]
```
Comment (lafriks) [inline modules/ssh/ssh.go:82]:
```go
		_, gitProtocol, _ = strings.Cut(env, "=")
```

Comment (Gusted) [reply modules/ssh/ssh.go:82]: strings.Cut was introduced in Go1.18, Gitea's minimum go version is still 1.17

Comment (jolheiser) [reply modules/ssh/ssh.go:82]: Gitea uses go 1.18 as of https://github.com/go-gitea/gitea/pull/19918

Comment (Gusted) [reply modules/ssh/ssh.go:82]: Huh? I was under the impression it that the minimum version is stable-1

Comment (delvh) [reply modules/ssh/ssh.go:82]: We decided to switch a bit earlier to 1.18.

As now 1.19 is out, we are defintely back on schedule.

Anyway, this discussion is probably a little out of place here...

Comment (Gusted) [reply modules/ssh/ssh.go:82]: https://github.com/go-gitea/gitea/pull/20638

**Pull request merged**